crates/orderbook/src/cow_subsidy.rs

```diff
@@ -24,66 +25,89 @@ impl Default for FixedCowSubsidy {
 
 #[async_trait::async_trait]
 impl CowSubsidy for FixedCowSubsidy {
+    /// On success returns a factor of how much of the usual fee an account has to pay based on the
+    /// amount of COW tokens it owns.
     async fn cow_subsidy_factor(&self, _: H160) -> Result<f64> {
         Ok(self.0)
     }
 }
 
-#[derive(Debug, Clone, Copy)]
-enum SubsidyTier {
-    None,
-    Full,
+/// Maps how many base units of COW someone must own at least in order to qualify for a given
+/// fee subsidy factor.
+#[derive(Clone, Debug, Default)]
+pub struct SubsidyTiers(BTreeMap<U256, f64>);
+
+impl std::str::FromStr for SubsidyTiers {
+    type Err = anyhow::Error;
+    fn from_str(serialized: &str) -> Result<Self, Self::Err> {
+        let mut tiers = BTreeMap::default();
+
+        for tier in serialized.split(',') {
+            let (threshold, fee_factor) = tier
+                .split_once(':')
+                .ok_or(anyhow::anyhow!("not enough arguments for subsidy tier"))?;
+
+            let threshold: f64 = threshold
+                .parse()
+                .map_err(|_| anyhow::anyhow!("can not parse threshold \"{}\" as f64", threshold))?;
+            let threshold = U256::from_f64_lossy(threshold * 1e18);
+
+            let fee_factor: f64 = fee_factor.parse().map_err(|_| {
+                anyhow::anyhow!("can not parse fee factor \"{}\" as f64", fee_factor)
+            })?;
+            if !(0.0..=1.0).contains(&fee_factor) {
+                anyhow::bail!("fee factor must be in the range of [0.0, 1.0]");
+            }
```

Comment:
Last `anyhow`-suggestion, I promise 😛. There is an `ensure!` macro which is like an `assert!` but returns an error instead of panicking.
```suggestion
            anyhow::ensure!(
                (0.0..=1.0).contains(&fee_factor),
                "fee factor must be in the range of [0.0, 1.0]"
            );
```